theia-extensions/viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -1,13 +1,23 @@
 import { spawn } from 'child_process';
+import { join } from 'path';
+import { existsSync } from 'fs';
 import { injectable } from 'inversify';
 import { PortBusy, TraceServerConfigService } from '../common/trace-server-config';
+import { FileOperationError, FileOperationResult } from '@theia/filesystem/lib/common/files';
 import treeKill = require('tree-kill');
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
     private processId: number;
 
     startTraceServer(path: string | undefined, port: number | undefined): Promise<string> {
-        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+        if (typeof path === 'undefined') {
+            throw new FileOperationError('The Trace Viwer path is not defined.', FileOperationResult.FILE_INVALID_PATH);
+        }
+        const serverPath = join(__dirname, '../../../../', path);
+        if (existsSync(serverPath) === false) {
+            throw new FileOperationError(`The Trace Viewer path (${serverPath}) was not found.`, FileOperationResult.FILE_NOT_FOUND);
+        }
+        const server = spawn(`${serverPath}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
```

Comment:
I suggest to keep this implementation as is (read values straight from the preferences) and let the user specify the exact path to the server.

Reply:
The usage of __dirname is mandatory if I want the application to work with an AppImage. Because by default the relative path is towards the location of the AppImage file, and not the mounted directory.